frontend/app/src/App.test.tsx

```diff
@@ -2383,5 +2445,31 @@ describe("App", () => {
         message: "random string",
       })
     })
+
+    it("properly handles TERMINATE_WEBSOCKET_CONNECTION and RESTART_WEBSOCKET_CONNECTION messages", () => {
+      const hostCommunicationMgr = prepareHostCommunicationManager()
+
+      fireWindowPostMessage({
+        type: "TERMINATE_WEBSOCKET_CONNECTION",
+      })
+
+      expect(getMockConnectionManager().disconnect).toHaveBeenCalled()
+
+      expect(hostCommunicationMgr.sendMessageToHost).toHaveBeenLastCalledWith({
+        type: "GUEST_READY",
+      })
+
+      fireWindowPostMessage({
+        type: "RESTART_WEBSOCKET_CONNECTION",
+      })
+
+      // This test isn't ideal, but since we can't inspect the state of our app
+      // using RTL, we have to check that some other side effect of calling
+      // initializeConnectionManager occurs.
```

Comment:
I know it sounds silly, but have we considered also have a `WEBSOCKET_CONNECTING` message? No one needs to know it, but it does handle this in between state well.

Reply:
Hm, I'd rather not add a new message type if we have no plans to actually use it. I figured out another way to write this test that doesn't rely on some almost unrelated side effect